modules/linear_boltzmann_solvers/executors/pi_keigen_smm.cc

```diff
@@ -0,0 +1,1122 @@
+    if ((bc.type == BoundaryType::ISOTROPIC) or (bc.type == BoundaryType::ARBITRARY))
+      throw std::logic_error("Only vacuum and reflective boundaries are valid for "
+                             "k-eigenvalue problems.");
+
+  const auto bcs = TranslateBCs(lbs_solver_.SweepBoundaries(),
+                                /*vacuum_bcs_are_dirichlet=*/false);
+
+  // Create the diffusion materials
+  const auto xs_map = PackGroupsetXS(
+    lbs_solver_.GetMatID2XSMap(), front_gs_.groups_.front().id_, front_gs_.groups_.back().id_);
+
+  // Create the appropriate solver
+  log.Log() << "Creating diffusion solver";
+  auto& diff_solver = diffusion_solver_;
+  if (diffusion_sdm_name_ == "pwld")
+  {
+    diff_solver = std::make_shared<DiffusionMIPSolver>(std::string(TextName() + "_SMM"),
+                                                       pwld,
+                                                       diff_uk_man,
+                                                       bcs,
+                                                       xs_map,
+                                                       lbs_solver_.GetUnitCellMatrices(),
+                                                       diffusion_verbose_);
+  }
+  else
+  {
+    diff_solver = std::make_shared<DiffusionPWLCSolver>(std::string(TextName() + "_SMM"),
+                                                        *pwlc_ptr_,
+                                                        diff_uk_man,
+                                                        bcs,
+                                                        xs_map,
+                                                        lbs_solver_.GetUnitCellMatrices(),
+                                                        diffusion_verbose_);
+  }
+
+  diff_solver->options.residual_tolerance = diffusion_l_abs_tol_;
+  diff_solver->options.max_iters = diffusion_l_max_its_;
+  diff_solver->options.verbose = diffusion_verbose_;
+  diff_solver->options.additional_options_string = diffusion_petsc_options_;
+  log.Log() << "Done creating diffusion solver.";
+
+  // Initialize the solver
+  log.Log() << "Initializing diffusion solver.";
+  diff_solver->Initialize();
+  mpi_comm.barrier();
+  log.Log() << "Done initializing diffusion solver.";
+
+  // Assemble the system
+  log.Log() << "Assembling diffusion system.";
+  std::vector<double> tmp;
+  if (diffusion_sdm_name_ == "pwld")
+    tmp.assign(pwld.GetNumLocalDOFs(diff_uk_man), 0.0);
+  else
+    tmp.assign(pwlc_ptr_->GetNumLocalAndGhostDOFs(diff_uk_man), 0.0);
+  diff_solver->AssembleAand_b(tmp);
+  log.Log() << "Done assembling diffusion system.";
+}
+
+void
+PowerIterationKEigenSMM::Execute()
+{
+  // Start transport power iterations
+  double k_eff_ell = k_eff_;
+  double k_eff_change = 1.0;
+
+  auto phi_ell = phi_old_local_;
+
+  int nit = 0;
+  while (nit < max_iters_)
+  {
+    // Compute the transport 1/k fission source
+    SetLBSFissionSource(phi_ell, /*additive=*/false);
```

Comment:
Remove `/*additive=*/`

Reply:
I actually thought it was a clever way of commenting something related to the specifics of the method. I would very much prefer to keep such comments.